internal/services/issuer/verifiable_credential.go

```diff
@@ -180,3 +189,98 @@ func (i *Issuer) VIN(vin string, tokenID *big.Int) (id string, err error) {
 
 	return id, tx.Commit()
 }
+
+func (i *Issuer) ADVinCredentialer(ctx goka.Context, msg interface{}) {
+	event, ok := msg.(*ADVinCredentialEvent)
+	if !ok {
+		i.log.Err(errors.New("unable to parse fingerprint event")).Msg("invalid payload")
+		return
+	}
+	currentIssuanceWeek := GetWeekNumForCron(time.Now())
+	observedVIN, err := services.ExtractVIN(event.Data)
+	if err != nil {
+		i.log.Info().Err(err).Str("userDeviceId", event.Subject).Msg("could not extract vin from payload")
+		return
+	}
+
+	ad, err := models.AutopiUnits(
+		models.AutopiUnitWhere.EthereumAddress.EQ(null.BytesFrom(common.FromHex(event.Subject))),
+		qm.Load(models.AutopiUnitRels.VehicleToken),
+	).One(ctx.Context(), i.DBS.DBS().Reader)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			i.log.Info().Err(err).Str("userDeviceId", event.Subject).Msg("no associated vehicle nft for device")
+			return
+		}
+		i.log.Info().Err(err).Str("userDeviceId", event.Subject).Msg("database failure retrieving user device")
+		return
+	}
+
+	tkn, ok := ad.VehicleTokenID.Int64()
+	if !ok {
+		i.log.Err(errors.New("unable to parse token id")).Msg("invalid token")
+	}
+
+	if val := ctx.Value(); val != nil {
+		record := val.(*VinEligibilityStatus)
+		if observedVIN == record.VIN && record.LatestEligibleRewardWeek != currentIssuanceWeek {
+			// TODO AE: also check that a vc hasn't been issued with a exp date GTE time.Now()?
+			_, err = i.VIN(observedVIN, big.NewInt(tkn))
+			record.LatestEligibleRewardWeek = currentIssuanceWeek
+			ctx.SetValue(record)
+			if err != nil {
+				i.log.Debug().Err(err).Str("userDeviceId", event.Subject).Str("vin", observedVIN).Int("issuanceWeek", currentIssuanceWeek).Msg("could not issue vin credential")
+				return
+			}
+		}
+		return
+	}
+
+	ud, err := models.UserDevices(
+		models.UserDeviceWhere.ID.EQ(ad.AutopiDeviceID.String),
```

Comment:
`AutopiDeviceID` is the identifier from AutoPi's paths. You probably want the attribute on the vehicle NFT.